### PR TITLE
Fix issue where package provider override isn't working when specifying a package_source

### PR DESCRIPTION
**Pull request by irvingpop**
Per the comments in https://github.com/chef/chef/issues/3487  -  passing a provider to the package resource is unreliable and causing problems for our customers that need to specify a `package_source`

This PR also adds tests for the `test::local` recipe and gets up to 100% test coverage :smile: 

cc: @jtimberman

### CODE REVIEW

**Comment by jtimberman** [comment]
Is this still required, given your last comment https://github.com/chef/chef/issues/3487?

**Comment by irvingpop** [comment]
Hey @jtimberman - Lamont chimed in with a much nicer implementation using `declare_resource` - and provided a more in-depth explanation about why *not* using `provider` is a good idea: https://github.com/chef/chef/issues/3487#issuecomment-110446611

I updated this PR to use `declare_resource` and it's much readable than before.   Also it has some awesome tests :)

I guess ultimately it comes down to how you feel about promoting the `declare resource` pattern over the `provider Chef::Package::Provider::Rpm` pattern.

File: libraries/helpers.rb [hunk 1]
```diff
@@ -39,9 +39,9 @@ def chef_server_ctl_command(pkg)
       ctl_cmds[pkg]
     end
 
-    def local_provider
-      return Chef::Provider::Package::Dpkg if node['platform_family'] == 'debian'
-      return Chef::Provider::Package::Rpm if node['platform_family'] == 'rhel'
+    def local_package_resource
+      return :dpkg_package if node['platform_family'] == 'debian'
+      return :rpm_package if node['platform_family'] == 'rhel'
```
Comment (lamont-granquist) [inline libraries/helpers.rb]: should probably return `:package` here by default, or else raise if you don't want to try to guess that.

**Comment by lamont-granquist** [comment]
The declare_resource bits LGTM :+1:

File: libraries/helpers.rb (hunk 1 above)
Comment (irvingpop) [inline libraries/helpers.rb]: good point, done

File: metadata.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 name 'chef-server-ingredient'
-version '0.3.2'
+version '0.3.3'
```
Comment (jtimberman) [inline metadata.rb]: Let's not increment versions in PR's :cake:

Comment (irvingpop) [inline metadata.rb]: sorry about that!  fixed.

**Pull request merged**